v2/brigadier/src/jobs.ts

```diff
@@ -20,33 +61,177 @@ export class Job {
+   * script authors have the option of a consistent style for creating
+   * and composing jobs and groups.)
+   * 
+   * @param name The name of the job
+   * @param image The OCI image reference for the primary container
+   * @param event The event that triggered the job
+   */
+  public static container(
+    name: string,
+    image: string,
+    event: Event
+  ): Job {
+    return new Job(name, image, event)
+  }
+
+  /**
+   * Specifies a Runnable that consists of sub-Runnables (such as jobs
+   * or concurrent groups) running one after another.
+   * A new Runnable is started only when the previous one completes.
+   * The sequence completes when the last Runnable has completed (or when any
+   * Runnable fails).
+   * 
+   * @param runnables The work items to be run in sequence
+   */
+  public static sequence(...runnables: Runnable[]): SerialGroup {
+    return new SerialGroup(...runnables)
+  }
+
+  /**
+   * Specifies a Runnable that consists of sub-Runnables (such as jobs
+   * or sequential groups) running concurrently.
+   * When run, all Runnables are started simultaneously (subject to
+   * scheduling constraints).
+   * The concurrent group completes when all Runnables have completed.
+   * 
+   * @param runnables The work items to be run in parallel
+   */
+  public static concurrent(...runnables: Runnable[]): ConcurrentGroup {
+    return new ConcurrentGroup(...runnables)
+  }
 }
 
 export class Container {
+  /** The OCI image that the container should run. */
   public image: string
+  /**
+   * Specifies under what conditions the host node should re-fetch an image
+   * that is already in its local cache. The permitted values are as IfNotPresent
+   * (the default) and Always.
+   */
   public imagePullPolicy = "IfNotPresent"
+  /** The working directory for the process running in the container. */
   public workingDirectory = ""
+  /**
+   * The command to run in the container. If not specified, the default entry point
+   * of the container is called.
+   * 
+   * Only the first element of the array is the actual command. Subsequent elements
+   * are treated as arguments. For example, a command of ["echo", "hello"] is
+   * equivalent to running 'echo hello'. A common convention is to use the command
+   * array for subcommands and the arguments array for argument values.
+   * 
+   * @example
+   * job.command = ["helm", "install"]
+   * job.arguments = ["stable/nginx", "-g"]
+   */
   public command: string[] = []
+  /**
+   * The arguments to pass to Container#command.  If the command includes arguments
+   * already, the arguments property are appended. A common convention is to use the command
+   * array for subcommands and the arguments array for argument values.
+   * 
+   * @example
+   * job.command = ["helm", "install"]
+   * job.arguments = ["stable/nginx", "-g"]
+   * 
+   */
   public arguments: string[] = []
+  /** Environment variables to set in the container. */
```

Comment:
A common thing that will occur is that users will inject project secrets into the environment variables here-- which is perfectly safe because on the back end we treat _all_ of these environment variables as if they might have been secrets.

Is it worth adding example usage? It looks like this:

```
job.primaryContainer.env.FOO = e.project.secrets.foo
```

Reply:
Added some wording and example, please check when pushed